saleor/webhook/transport/asynchronous/transport.py

```diff
@@ -196,7 +207,11 @@ def trigger_webhooks_async(
             raise NotImplementedError("No payload was provided for regular webhooks.")
 
         with allow_writer():
-            payload = EventPayload.objects.create(payload=data)
+            payload = EventPayload.objects.create()
```

Comment:
Shouldn't be in `transaction.atomic`?
What will happen if `payload.payload_file.save(` raise an exception? It communicates with external storage so it could be possible, right?

Reply:
If we age that transaction is required here we should also add it to similar code inside `create_deliveries_for_subscriptions` of this file and similar code inside `saleor/webhook/transport/synchronous/transport.py`

Reply:
Then also maybe we can add some small helper function as there will be three things:
- transaction
- create empty payload instance
- save file

Reply:
+ also error handling in case the file save goes wrong, as Maciek pointed.